pyincore/semanticservice.py

```diff
@@ -0,0 +1,178 @@
+        self.client = client
+        self.base_url = urljoin(client.service_url, "semantics/api/types")
+
+    @staticmethod
+    def return_http_response(http_response):
+        try:
+            http_response.raise_for_status()
+            return http_response
+        except requests.exceptions.HTTPError:
+            logger.error(
+                "A HTTPError has occurred \n"
+                + "HTTP Status code: "
+                + str(http_response.status_code)
+                + "\n"
+                + "Error Message: "
+                + http_response.content.decode()
+            )
+            raise
+        except requests.exceptions.ConnectionError:
+            logger.error(
+                "ConnectionError: Failed to establish a connection with the server. "
+                "This might be due to a refused connection. "
+                "Please check that you are using the right URLs."
+            )
+            raise
+        except requests.exceptions.RequestException:
+            logger.error(
+                "RequestException: There was an exception while trying to handle your request. "
+                "Please go to the end of this message for more specific information about the exception."
+            )
+            raise
+
+    def get_all_semantic_types(
+        self,
+        hyperlink: bool = False,
+        order: str = "asc",
+        skip: int = 0,
+        limit: int = 50,
+        detail: bool = False,
+        save_json: bool = False,
+        json_path: str = None,
+        timeout: Tuple[int, int] = (30, 600),
+        **kwargs,
+    ) -> Union[list, dict]:
+        """Get all semantic types.
+
+        Args:
+            hyperlink (bool): If True, return a list of hyperlinks to the semantic types.
+            order (str): Order of the semantic types. Can be 'asc' or 'desc'.
+            skip (int): Number of semantic types to skip.
+            limit (int): Number of semantic types to return.
+            detail (bool): If True, return detailed information about the semantic types.
+            save_json (bool): If True, save the json response to a file.
+            json_path (str): Path to save the json response.
+            timeout (tuple): Timeout for the request in seconds.
+                            First value is the connection timeout, second value is the read timeout.
+            **kwargs: Arbitrary keyword arguments.
+
+        Returns:
+            list: List of semantic types.
+
+        """
+
+        url = self.base_url
+        payload = {
+            "hyperlink": hyperlink,
+            "order": order,
+            "skip": skip,
+            "limit": limit,
+            "detail": detail,
+        }
+        response = self.client.get(
+            url, params=payload, timeout=timeout, **kwargs
+        )
+        data = self.return_http_response(response).json()
+
+        if save_json:
+            # if json_path is None: then save to the default location (incore cache folder)
```

Comment:
Save to cache will be very hard for user to find. Why not just pass empty path so it will save at where user run this method?

Reply:
ahh, good point. I'll do that.